App/index.js

```diff
@@ -12,34 +12,47 @@ function saveToLocalStorage(){
 function loadFromLocalStorage(){
     const reviewListString = localStorage.getItem('reviewList');
     const reviewList = JSON.parse(reviewListString);
-    return reviewList;
+    return reviewList ?? [];
 }
 
+
+
 function renderReviews() {
     //set the appropiate star behavior
     const stars = document.getElementsByClassName("stars");
     setStars(selectedStar,stars);
-    for (let i = 0; i < stars.length; i++){
-        stars[i].onmouseover = () => {
-            stars[i].src = "star_full.png";
-            for (let j = 0; j < i; j++){
-                stars[j].src = "star_full.png";
+
```

Comment:
Možda još bolja ideja: da suziš scope, pa da evente ne sluša čitav dokument, već samo ovaj stars wrapper. Nakon ovog onda umjesto dodavanja event listenera na dokument, staviš ih na starContainer
```suggestion

const starContainer = document.querySelector(".starContainer");

```